src/libmongoc/src/mongoc/mongoc-client-session.c

```diff
@@ -1621,6 +1692,26 @@ _mongoc_client_session_pin (mongoc_client_session_t *session,
    session->server_id = server_id;
 }
 
+void
+_mongoc_client_session_set_snapshot_time (mongoc_client_session_t *session,
+                                          uint32_t t,
+                                          uint32_t i)
+{
+   BSON_ASSERT (session);
```

Comment:
Since I think calling this when a snapshot time is already set is a logic error:

```suggestion
   BSON_ASSERT (session);
   BSON_ASSERT (!session->snapshot_time_set);
```

Reply:
Added.